### DISCUSSION TITLE
Is there a hook that lets me filter the original long url that the user enters?

### DISCUSSION THREAD

**Discussion by Spatino19** [YOURLS/YOURLS#2890] [Q&A] [answered]
I am trying to make a plugin and i checked the hook list several times and tried different filters but none have worked.

I want to modify the original long url the user enters. The one url that the short url will redirect to

So with a filter I can add anything as a prefix to the original url
For example, "example.com/" can is the prefix and the user enters "text" then with the filter it creates the long url "example.com/text" that will be stored in the database and later used as the redirect url for the short url

**Comment by ozh** [comment, marked as answer]
The function called when shortening a new URL is [`yourls_add_new_link()`](https://github.com/YOURLS/YOURLS/blob/master/includes/functions-shorturls.php#L28)

Here, there isn't indeed a straightforward `$url = yourls_apply_filter("some_hook", $url);`
That might be added, for instance after the [`yourls_do_action( 'pre_add_new_link', $url, $keyword, $title );`](https://github.com/YOURLS/YOURLS/blob/master/includes/functions-shorturls.php#L58) to filter the submitted URL, keyword and title.

An easy alternative is to hook to one of the numerous filters already called within this function or within functions called by this function (for instance `yourls_sanitize_url()` ...)

**Reply by Spatino19** [reply]
Alright. Hopefully that filter gets added soon. Thanks for giving me a possible alternative

**Reply by dgw** [reply]
When Ozh says it "might be added", that's an invitation to submit a pull request that adds the hook you need, if the alternatives don't work out. 😺

**Reply by luanfernandes** [reply]
Can someone post an example on how it can be done currently? I'm also trying to change the long url before going to the database. I already got it working using pre_redirect but I can't make it work otherwise.

I tried with: 
yourls_add_action('pre_add_new_link, 'my_function')
yourls_add_filter('add_new_link','my_function')

and so on...

I need some light @ozh !

**Reply by ozh** [reply]
The first bit in `yourls_add_new_link()` that modifies the URL is `$url = yourls_sanitize_url( $url );`, so why don't you use a filter in this function ? Easy there to check if the URL is just "text" and return "example.com/text" as in your example

**Reply by luanfernandes** [reply]
so I would use

> yourls_add_filter( 'yourls_sanitize_url', 'my_function' );
> // Our custom function that will be triggered when the event occurs
> function my_function( $args ) {
>	$old_url = $args[0];

for example?

**Reply by ozh** [reply]
No... [`yourls_sanitize_url()`](https://github.com/YOURLS/YOURLS/blob/af30597777b6a51f8cc0c2d75e9b32bace07fe65/includes/functions-formatting.php#L108) returns `yourls_apply_filter( 'sanitize_url', $url, $unsafe_url )`, so you want to return the $unsafe_url that was first passed in

**Reply by luanfernandes** [reply]
oh I see. Well, still no luck with it.. hopefully I will figure it out soon. Thanks.

**Reply by Spatino19** [reply]
I was just trying to work on it. So, should it be something like this?

```
yourls_add_filter('yourls_sanatize_url', 'my_own_function' );

function my_own_function($unsafe_url) {
	$unsafe_url = "prefix" . $unsafe_url;
	
	return $unsafe_url;
}
```
I am just starting to learn how to code in general so I have no experience at all so I could be making newbie mistakes

**Reply by ozh** [reply]
I suggest you start with
- reading https://github.com/YOURLS/YOURLS/wiki/How-to-make-Plugins
- checking a few simple plugins on https://github.com/YOURLS/awesome-yourls to get the grasps of it
- eventually checking https://github.com/YOURLS/YOURLS/wiki/Possible-With-A-Plugin#find-a-plugin-coder if it's not within your reach

**Reply by Spatino19** [reply]
@luanfernandes were you able to do it? I have been trying for days and I can't add a prefix. I have been reading the tutorials and I have been using a similar structure to [yourls-keyword-prefix](https://github.com/jangrewe/yourls-keyword-prefix/blob/master/plugin.php) but for some reason there's no prefix yet.

**Reply by luanfernandes** [reply]
No I was not. I didn't have much time to try other stuff yet though
